codepropertygraph/src/main/scala/io/shiftleft/utils/StringInterner.scala

```diff
@@ -0,0 +1,34 @@
+package io.shiftleft.utils
+
+import gnu.trove.map.hash.THashMap
+
+trait StringInterner {
+  def intern(s: String): String
+}
+
+object StringInterner {
+  val DefaultMaxStringLength: Int = 1024
+  val DefaultInitialSize: Int = 64 * 1024
+
+  val noop: StringInterner = new StringInterner {
+    override def intern(s: String): String = s
+  }
+  def makeStrongInterner(maxStringLength: Int = DefaultMaxStringLength,
```

Comment:
why not just a regular class with a constructor? would have avoided the typo as well :)